Updated Readme.md (#2)
perjo927/gitbot: README.md

# Gitbot

GitBot is a PowerShell script designed to automate common Git operations and simplify your Git workflows in Windows OS. You can click here ➡️ [Medium Article](https://medium.com/@djpjgj/make-your-github-stats-great-again-a830cb6e3428) to know further about the usecase.

## User Guide

### Initial Setup

1. Ensure that you have PowerShell or Command prompt installed on your system.

> 2. Clone the GitBot repository to your local machine using the following command after navigationg to the directory of your choice:

Reviewer: ```suggestion
2. Clone the GitBot repository to your local machine using the following command after navigating to the directory of your choice:
```

   - Feel free to explore the script and customize it according to your specific requirements.

3. Save the modified script.

4. Open the `schedule_task.ps1` script in a text editor.

5. Modify the `$trigger` variable to adjust the desired schedule. By default, it is set to trigger the GitBot script daily at 6 PM. You can customize the schedule by modifying this variable.

6. Save the modified script.

### Execution

1. Open a PowerShell terminal or Command prompt Window and navigate to the `GitBot repository folder`.

2. Execute the GitBot script using the following command:

   ```
   powershell -ExecutionPolicy Bypass -File gitbot.ps1 -path <path-to-your-repo>
   ```

   Replace `<path-to-your-repo>` with the path to your target repository which you have newly created and has words.txt in it. This will initialize the GitBot process.

3. GitBot will create a new branch with a random name based on the words in the `words.txt` file, add a JavaScript file with content, commit the changes, push them to the repository, merge the branch into the default branch, and clean up the branch.

4. Repeat step 2 as needed to run GitBot manually.

### Scheduling

> 1. Open a PowerShell terminal or Command prompt Windowand navigate to the `GitBot repository folder` again.

Reviewer: ```suggestion
1. Open a PowerShell terminal or Command prompt window and navigate to the `GitBot repository folder` again.
```

2. Execute the `schedule_task.ps1` script using the following command:

   ```
   powershell -ExecutionPolicy Bypass -File schedule_task.ps1 -path <path to gitbot.ps1>
   ```
   
   Replace `<path-to-your-repo>` with the path to the directory where `gitbot.ps1` is present.

   This will create a scheduled task in Windows Task Scheduler, triggering the GitBot script based on the configured schedule.

> 3. Repeat the above  2 steps if you want to reschedule the GitBot process.

Reviewer: ```suggestion
3. Repeat the above 2 steps if you want to reschedule the GitBot process.
```
